FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/portable/NetworkInterface/STM32Hxx/NetworkInterface.c

```diff
@@ -0,0 +1,557 @@
+		buffer.  The Ethernet buffer is therefore no longer needed, and must be
+		freed for re-use. */
+		vReleaseNetworkBufferAndDescriptor( pxDescriptor );
+	}
+
+	if (status == HAL_OK)
+	{
+		xResult = pdPASS;
+	}
+	else
+	{
+		xResult = pdFAIL;
+	}
+
+	return xResult;
+}
+
+
+/*******************************************************************************
+			           END Network Interface API Functions
+*******************************************************************************/
+
+
+
+/*******************************************************************************
+			           Network Interface Static Functions
+*******************************************************************************/
+
+static void prvMACBProbePhy( void )
+{
+	/* Bind the write and read access functions. */
+	vPhyInitialise( &xPhyObject,
+					( xApplicationPhyReadHook_t ) ETH_PHY_IO_ReadReg,
+					( xApplicationPhyWriteHook_t ) ETH_PHY_IO_WriteReg );
+	/* Poll the bus for all connected PHY's. */
+	xPhyDiscover( &xPhyObject );
+	/* Configure them using the properties provided. */
+	xPhyConfigure( &xPhyObject, &xPHYProperties );
+}
+
+static void prvEthernetUpdateConfig( BaseType_t xForce )
+{
+	ETH_MACConfigTypeDef MACConf;
+	uint32_t speed = 0, duplex =0;
+
+	FreeRTOS_printf( ( "prvEthernetUpdateConfig: LS mask %02lX Force %d\n",
+			xPhyObject.ulLinkStatusMask,
+			( int )xForce ) );
+
+	if( ( xForce != pdFALSE ) || ( xPhyObject.ulLinkStatusMask != 0 ) )
+	{
+		/* Restart the auto-negotiation. */
+		xPhyStartAutoNegotiation( &xPhyObject, xPhyGetMask( &xPhyObject ) );
+
+		/* Configure the MAC with the Duplex Mode fixed by the
+		   auto-negotiation process. */
+		if( xPhyObject.xPhyProperties.ucDuplex == PHY_DUPLEX_FULL )
+		{
+			duplex = ETH_FULLDUPLEX_MODE;
+		}
+		else
+		{
+			duplex = ETH_HALFDUPLEX_MODE;
+		}
+
+		/* Configure the MAC with the speed fixed by the
+		   auto-negotiation process. */
+		if( xPhyObject.xPhyProperties.ucSpeed == PHY_SPEED_10 )
+		{
+			speed = ETH_SPEED_10M;
+		}
+		else
+		{
+			speed = ETH_SPEED_100M;
+		}
+
+		/* Get MAC and configure it */
+		HAL_ETH_GetMACConfig(&xEthHandle, &MACConf);
+		MACConf.DuplexMode = duplex;
+		MACConf.Speed = speed;
+		HAL_ETH_SetMACConfig(&xEthHandle, &MACConf);
+
+		/* Restart MAC interface */
+		HAL_ETH_Start_IT(&xEthHandle);
+	}
+	else
+	{
+		/* Stop MAC interface */
+		HAL_ETH_Stop_IT( &xEthHandle );
+	}
+}
+
+static void prvEMACHandlerTask( void *pvParameters )
+{
+BaseType_t xResult;
+TickType_t uxPollInterval = pdMS_TO_TICKS( 1u );
+
+	( void ) pvParameters;
+
+	if( uxPollInterval < 1u )
+	{
+		uxPollInterval = 1u;
+	}
+
+	for( ;; )
+	{
+		/* Wait for the Ethernet MAC interrupt to indicate that another packet
+		has been received.  The task notification is used in a similar way to a
+		counting semaphore to count Rx events, but is a lot more efficient than
+		a semaphore. */
+
+		/* ulTaskNotifyTake( pdFALSE, portMAX_DELAY ); */
```

Comment:
Yes, I commented and replaced it. The driver may not block for ever because it must check the PHY status from time to time, even if there are no packets.

Reply:
My issue was this: Is there a point to keep this comment? How would it help someone?
We should either add a comment about it or remove it completely.
Thanks

Reply:
My issue was this: Is there a point to keep this comment? How would it help someone?
We should either add a comment about it or remove it completely.
Thanks